### PR TITLE
Hook playwright tests up to a local Convex backend

### PR DESCRIPTION
**Pull request by dowski**
This takes the tests from #104 and sets them up to run against a local Convex backend. Given a local checkout of `convex-backend` and properly configured [build dependencies](https://github.com/get-convex/convex-backend?tab=readme-ov-file#dependencies), a command like the following will spin up a local backend instance, configure it for auth and run the tests.

```
CONVEX_LOCAL_BACKEND_PATH=/Users/christian/Development/Convex/convex-backend npm run test
```

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test-nextjs/e2e-tests/README.md [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+
+1. Clone [convex-backend](https://github.com/get-convex/convex-backend)
+
+1. Follow the instructions in its [README](https://github.com/get-convex/convex-backend/blob/main/README.md) to get it building
+
+1. From the `test-nextjs` directory, run:
+
+```
+CONVEX_LOCAL_BACKEND_PATH=/path/to/your/convex-backend npm run test
+```
```
Comment (xixixao) [inline test-nextjs/e2e-tests/README.md:16]: Or use the built image? (easier)

Comment (xixixao) [reply test-nextjs/e2e-tests/README.md:16]: Total off-topic: I don't know if Prettier does it or what, but having the markdown lists be 1. (which works when rendered) makes it hard to read the source (which most devs will do).

Comment (dowski) [reply test-nextjs/e2e-tests/README.md:16]: Thanks for the comments.

I'm going to leave the instructions as-is for now as they work and it's a one-time setup cost for someone who wants to run the end-to-end tests.

As for the Markdown list format, I just went with the convention that was already in the README.

**Review by sshader — APPROVED** [review]

**Comment by dowski** [comment]
Thanks for reviewing!

**Pull request merged**